### PR TITLE
fix: (CXSPA-8048) - Remove narration of Updating Cart in AddedToCartDialog component

### PR DESCRIPTION
**Pull request by petarmarkov9449**
Closes: [CXSPA-8048](https://jira.tools.sap/browse/CXSPA-8048)

Remove narration of "Updating Cart ..." in AddedToCartDialog component

### CODE REVIEW

**Review by Pio-Bar — DISMISSED** [review]

**Review by Zeyber — DISMISSED** [review]

**Comment by developpeurweb** [comment]
@petarmarkov9449 for some reason, it seems we have a regression here. As per screenshot, the current branch narrates the _Page Title_ as the dialog's title, as opposed to `develop` branch where the "Updating Cart" narration is set as the dialog's title. 

**Both are inaccurate.** The narration of the dialog's title must match whatever is visible as the dialog's title. E.g.,
- Visible: "Item(s) added to your cart" = Narration: "Item(s) added to your cart **_dialog_**"
- Visible: "This item was already in your cart. The quantity was updated" = Narration: "This item was already in your cart. The quantity was updated **_dialog_**"

Basically, if hidden narrations such as "Loading..." or "Updating Cart..." are getting in the way of matching the visible title with audible one, then please remove those hidden narrations.

current branch
m.
develop

**Review by developpeurweb — CHANGES_REQUESTED** [review]
Visual and Audible Dialog's Title should match. Please see previous comments and screenshots.

**Review by developpeurweb — DISMISSED** [review]

**Comment by developpeurweb** [comment]
@petarmarkov9449 Works well now. Please have a developer approve your latest code changes. Thanks!

8048 fixed

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html [hunk 1]
```diff
@@ -132,7 +144,12 @@
   <ng-template #loading>
     <div class="cx-dialog-body modal-body">
       <div class="cx-dialog-row">
-        <div class="col-sm-12"><cx-spinner></cx-spinner></div>
+        <div class="col-sm-12" *cxFeature="'a11yUpdatingCartNoNarration'">
```
Comment (Pio-Bar) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: We don't consider the use of aria attributes breaking. 🙂

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts [hunk 2]
```diff
@@ -170,6 +178,7 @@ describe('AddedToCartDialogComponent', () => {
           useClass: MockRoutingService,
         },
         { provide: LaunchDialogService, useClass: MockLaunchDialogService },
+        { provide: FeatureConfigService, useClass: MockFeatureConfigService },
```
Comment (Pio-Bar) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts]: Is there a reason to provide this service? It doesn't look like it is used in the component.
Since the MockFeatureDirective is already declared that should cover your new feature flag in the template.

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts [hunk 3]
```diff
@@ -247,20 +257,12 @@ describe('AddedToCartDialogComponent', () => {
...
 
-  it('should display loading placeholder', () => {
-    component.loaded$ = of(false);
+  it('should shouw item added to your cart dialog title message in case a11yUpdatingCartNoNarration is enabled', () => {
+    spyOn(featureConfigService, 'isEnabled').and.returnValue(true);
     fixture.detectChanges();
     expect(
       el.query(By.css('.cx-dialog-title')).nativeElement.textContent.trim()
-    ).toEqual('addToCart.updatingCart');
-    expect(el.query(By.css('cx-spinner')).nativeElement).toBeDefined();
```
Comment (Pio-Bar) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts:256]: Are we no longer testing for the spinner presence? It looks like its still displayed?

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts [hunk 4]
```diff
@@ -247,20 +257,12 @@ describe('AddedToCartDialogComponent', () => {
     });
   });
 
-  it('should display loading placeholder', () => {
-    component.loaded$ = of(false);
+  it('should shouw item added to your cart dialog title message in case a11yUpdatingCartNoNarration is enabled', () => {
```
Comment (Pio-Bar) [inline feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts]: We only test the feature with the feature flag enabled, since the change will eventually become permanent. Let's test it as if it already were.

**Review by Pio-Bar — CHANGES_REQUESTED** [review]

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts (hunk 4 above)
Comment (petarmarkov9449) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts]: I've just updated it.

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts (hunk 3 above)
Comment (petarmarkov9449) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts:256]: I've just updated it.

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts (hunk 2 above)
Comment (petarmarkov9449) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.spec.ts]: I've just updated it.

File: feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html (hunk 1 above)
Comment (petarmarkov9449) [reply feature-libs/cart/base/components/added-to-cart-dialog/added-to-cart-dialog.component.html]: I've just updated it.

**Review by Pio-Bar — APPROVED** [review]

**Review by Zeyber — APPROVED** [review]

**Review by developpeurweb — APPROVED** [review]
